lib/livebook_web/live/session_live.ex

```diff
@@ -1116,8 +1164,33 @@ defmodule LivebookWeb.SessionLive do
           socket
 
         {:ok, section} ->
-          push_patch(socket,
-            to: ~p"/sessions/#{socket.assigns.session.id}/delete-section/#{section.id}"
+          section_id = section.id
+          first_section? = hd(socket.private.data.notebook.sections).id == section.id
+
+          on_confirm = fn socket, %{"delete_cells" => delete_cells} ->
+            Livebook.Session.delete_section(socket.assigns.session.pid, section_id, delete_cells)
+            socket
+          end
+
+          assigns = %{section_name: section.name}
+
+          description = ~H"""
+          Are you sure you want to delete this section - <span class="font-semibold">“<%= @section_name %>”</span>?
+          """
+
+          confirm(socket, on_confirm,
+            title: "Delete section",
+            description: description,
+            confirm_text: "Delete",
+            confirm_icon: "delete-bin-6-line",
+            options: [
+              %{
+                name: "delete_cells",
+                label: "Delete all cells in this section",
+                default: first_section?,
+                disabled: first_section?
+              }
+            ]
```

Comment:
When deleting a section we ask the user "Delete all cells in this section" and to do that we used a whole separate URL and modal. Now that we needed a similar option when deleting a file, I added this to the generic confirm we have, and now that it's all in Elixir it was actually very little code :D